src/python/pants/backend/jvm/tasks/jvm_task.py

```diff
@@ -74,14 +75,21 @@ def classpath(self, targets, classpath_prefix=None, classpath_product=None, excl
 
   def preferred_jvm_distribution_for_targets(self, targets, jdk=False, strict=False):
```

Comment:
It feels like this should be squared with `get_runtime_platform_for_target` in some way: if they are independent for a good reason, it would be good for them to reference one another and explain why. In particular: it's not clear why this is doing "simple" runtime detection and not falling back to the default as in the other case.

Reply:
Hm. I'll try pulling some of it out. I think the platform collection wrangling could live in JvmPlatform instead.

Reply:
oh right, it's setup that way to prevent a cycle between JvmTarget / RuntimePlatformMixin and JvmPlatform.

I could call get_runtime_platform_for_target from there, but I don't think it buys much because the call to `target.runtime_platform` just calls that itself.